bluesky/run_engine.py

```diff
@@ -132,6 +136,11 @@ class RunEngine:
         completely up to the user. The function's return value is
         ignored.
 
+    scan_id_source : callable, optional
+        a function that will be use to calculate scan_id. Defaultis increment
```

Comment:
Very good start with the docstring. Thanks for remembering to include this. Some minor copy editing:
* will be use -> will be used
* Defaultis is missing a space, and should say "Default is to increment"
* bespoken -> bespoke

I don't know if I would say a "complex" scan_id. Maybe it would be better to say something like "get a scan_id from any source".

Also, when we expected a callable, we document what the expected signature is -- meaning, what arguments it will be passed when it is called. See  the docstring for `md_validator` just above for a good example.

Reply:
My mistake.
Updated!